pkg/render/apiserver.go

```diff
@@ -838,7 +838,8 @@ func (c *apiServerComponent) apiServerDeployment() *appsv1.Deployment {
 					Containers: []corev1.Container{
 						c.apiServerContainer(),
 					},
-					Volumes: c.apiServerVolumes(),
+					Volumes:  c.apiServerVolumes(),
+					Affinity: podaffinity.NewPodAntiAffinity(name, rmeta.APIServerNamespace(c.installation.Variant)),
```

Comment:
I think we should only add this affinity if replicas is > 1. That way we can ensure no Affinity will be set if replicas is specified as 1.